test/ueberauth/strategy/github_test.exs

```diff
@@ -0,0 +1,103 @@
+defmodule Ueberauth.Strategy.GithubTest do
+  use ExUnit.Case
+
+  import Plug.Conn, only: [put_private: 3]
+
+  alias Ueberauth.Strategy.Github
+  alias Plug.Conn
+
+  setup do
+    Application.put_env(:ueberauth, Ueberauth.Strategy.Github.OAuth, [], persistent: true)
+    Application.put_env(:ueberauth_github, :oauth2_client, TestOAuthClient, persistent: true)
+    Application.put_env(:ueberauth_github, :oauth2_module, TestOAuth, persistent: true)
+    Application.put_env(:ueberauth_github, :uid_field, :login, persistent: true)
+  end
+
+  def auth_conn(test_params, url) do
+    %Conn{params: %{"scope" => "user,public_repo,gist"}}
+    |> Plug.Adapters.Test.Conn.conn("GET", url, test_params)
+    |> put_private(:ueberauth_request_options, [options: [:oauth2_module, Ueberauth.Strategy.Github.OAuth]])
+    |> put_private(:github_token, %TestOAuthToken{access_token: "foobar"})
+  end
+
+  test "handle_request" do
```

Comment:
I'd also like to write tests for this and handle_callback's error match functions, but havent yet